apm-agent-plugins/apm-spring-webmvc/apm-spring-webmvc-spring5/src/main/java/co/elastic/apm/agent/springwebmvc/Spring5ExceptionHandlerInstrumentation.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.springwebmvc;
+
+import co.elastic.apm.agent.servlet.Constants;
+import co.elastic.apm.agent.servlet.adapter.JavaxServletApiAdapter;
+import net.bytebuddy.asm.Advice;
+
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+
+public class Spring5ExceptionHandlerInstrumentation extends AbstractSpringExceptionHandlerInstrumentation {
+
+    @Override
+    public Constants.ServletImpl servletImpl() {
+        return Constants.ServletImpl.JAVAX;
+    }
+
+    public static class AdviceClass extends AbstractSpringExceptionHandlerInstrumentation.AdviceClass {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static void captureException(@Advice.Argument(0) @Nullable HttpServletRequest request,
+                                            @Advice.Argument(3) @Nullable Exception e) {
+            onEnter(JavaxServletApiAdapter.get(), request, e);
```

Comment:
[minor] given the adapter is only used to set an attribute which is only a single line, we could probably make this simpler by directly calling the adapter and avoid the advice inheritance and indirection.

Reply:
Good for me with the helper + not inline the implementation, we get the helper+method name to provide relevant naming which helps to understand once I'll have forgotten about this conversation :-).